lib/runtime-core/src/state.rs

```diff
@@ -854,5 +946,36 @@ pub mod x64 {
                 X64Register::XMM(x) => RegisterIndex(x as usize + 16),
             }
         }
+
+        pub fn from_dwarf_regnum(x: u16) -> Option<X64Register> {
+            Some(match x {
+                0 => X64Register::GPR(GPR::RAX),
+                1 => X64Register::GPR(GPR::RDX),
+                2 => X64Register::GPR(GPR::RCX),
+                3 => X64Register::GPR(GPR::RBX),
+                4 => X64Register::GPR(GPR::RSI),
+                5 => X64Register::GPR(GPR::RDI),
+                6 => X64Register::GPR(GPR::RBP),
+                7 => X64Register::GPR(GPR::RSP),
+                8 => X64Register::GPR(GPR::R8),
+                9 => X64Register::GPR(GPR::R9),
+                10 => X64Register::GPR(GPR::R10),
+                11 => X64Register::GPR(GPR::R11),
+                12 => X64Register::GPR(GPR::R12),
+                13 => X64Register::GPR(GPR::R13),
+                14 => X64Register::GPR(GPR::R14),
+                15 => X64Register::GPR(GPR::R15),
+
+                17 => X64Register::XMM(XMM::XMM0),
+                18 => X64Register::XMM(XMM::XMM1),
+                19 => X64Register::XMM(XMM::XMM2),
+                20 => X64Register::XMM(XMM::XMM3),
+                21 => X64Register::XMM(XMM::XMM4),
+                22 => X64Register::XMM(XMM::XMM5),
+                23 => X64Register::XMM(XMM::XMM6),
+                24 => X64Register::XMM(XMM::XMM7),
+                _ => return None,
```

Comment:
This applies to LLVM generated code too, right? Any reason to believe LLVM will never use XMM8--XMM31? YMM? ZMM/K (AVX-512)? I'm willing to say that we will never use x87 floating point stack or MMX registers.

Reply:
Good point. Will add XMM8-XMM31 in. Since all values tracked by the stack map are not larger than 64 bits (SIMD is an exception, but can fix it later), saving just XMM8-31 should be enough?

Reply:
If it works then it's fine for now, but I can't promise that LLVM won't pack four 64-bit values together in a YMM register for example. It could if it wanted to, it just doesn't seem to want to.

What goes wrong if LLVM uses registers that aren't listed here? a) Will it cause problems at runtime or do we just realize we can't tier-switch at this point and ignore it? b) Is there a way I can test whether other registers are used, starting with a given Wasm file?

Reply:
If the LLVM values "semantically" are at most 64 bits, then the equivalent runtime locations are unlikely to be larger than 64 bits - I don't think there's a way to represent in stackmap a value with different sizes between LLVM value and runtime location. Also, with the presence of stackmap some LLVM optimizations won't apply.

Reply:
The compiler will panic if this function returns None, indicating a bug.